lib/proxy/authentication.rb

```
@@ -0,0 +1,47 @@
+module Proxy::Authentication 
+  class Chef
+    require 'chef'
+    require 'digest/sha2'
+    require 'base64'
+    require 'openssl'
+    require 'helpers'
+
+    def verify_signature_request(client_name,signature,body)
+        #We need to retrieve node public key
+        #to verify signature
+        chefurl = SETTINGS.chefserver_url
+        client_name = SETTINGS.smartproxy_clientname
+        key = SETTINGS.smartproxy_privatekey
+
+        rest = ::Chef::REST.new(chefurl,client_name,key)
+        begin
+          public_key = OpenSSL::PKey::RSA.new(rest.get_rest("/clients/#{client_name}").public_key)
+        rescue
```

Here you should at least log the trace of exception. Also you should specify the exception class you are rescuing, these are hard to debug later since it swallow every exception. And I'm not sure what kind of error do you expect here.